queue.rb

```diff
@@ -0,0 +1,49 @@
+class Queue
+  attr_reader :size
+
+  def initialize
+    @queue = []
+    @size = 0
+  end
+
+  def enqueue(item)
+    @queue += [item]
+    @size += 1
+  end
+
+  def dequeue
+    dequeued = @queue[0]
+    @queue = @queue[1..(size-1)]
+    @size -= 1
+    return dequeued
+  end
+
+  def peek_front
+    @queue[0]
+  end
+
+  def peek_last
+    @queue[(size-1)]
```

Comment:
Unless you're just trying to not use any of the array helpers? Which totally makes sense. Otherwise
`@queue[-1]`